runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt

```diff
@@ -54,6 +54,7 @@ public inline class Worker @PublishedApi internal constructor(val id: Int) {
          *
          * @param pointer value returned earlier by [Worker.asCPointer]
          */
+        @Deprecated("Use kotlinx.cinterop.StableRef instead", level = DeprecationLevel.WARNING)
```

Comment:
Worked is an inline class over an Int, so intention of this API was to pass integer worker id as `COpaquePointer` without boxing. Stable ref to the box (which would be implicitly created)  is different operation, so I'd suggest either keep this operation as is, or avoid suggestion.

Reply:
> Worked is an inline class over an Int

Exactly. And this blocks us from making `.name` work for terminated workers too, as I mentioned in the commit message.

> so intention of this API was to pass integer worker id as COpaquePointer without boxing

I'm not sure this operation is actually required.

> Stable ref to the box (which would be implicitly created) is different operation

Sure, but it should be enough for real use cases.

> keep this operation

Not an option, since this operation is harmful, at least for the reason described above.

> or avoid suggestion.

Not an option, we should provide an alternative for deprecated operations.

Reply:
This will not change behavior of `name` in any sense, and as described in https://jetbrains.slack.com/archives/C0NKC62TV/p1584618031103200?thread_ts=1584613014.099300&cid=C0NKC62TV changing `name` behavior requires to change overall approach to the `Worker` API (make it object oriented and holding, not resource oriented). Worker API is modeled  after POSIX `pthread_t` and file/sockets descriptors, changing `*CPointer` API won't affect this design.

Reply:
> This will not change behavior of name in any sense

Indeed. But AFAIK, this API is the only user-visible thing blocking from making `Worker` an object. 

> changing name behavior requires to change overall approach to the Worker API (make it object oriented and holding, not resource oriented). 

I don't think so. Making `Worker` an object (holding `id` and e.g. `name`) and maintaining a map fro `id` to `Worker` internally should fix the problem.

Reply:
(Not sure if the map from `id` to `Worker` is required, it may be enough to have a worker-local variable for this).